addons/controllers/tanzuclusterbootstrap_controller.go

```diff
@@ -535,7 +542,12 @@ func (r *TanzuClusterBootstrapReconciler) updateValuesFromProvider(cluster *clus
 				UID:        cluster.UID,
 			},
 		})
-		newProvider.SetName(fmt.Sprintf("%s-%s-package", cluster.Name, packageShortName))
+		// Add cluster and package labels to cloned providers
+		newProvider.SetLabels(map[string]string{
```

Comment:
Same here. Do we need to preserve the labels added by others?

Reply:
Done